### ISSUE TITLE
Clean up javadoc for stricter Java 11

### ISSUE THREAD

**Issue by headius** [jruby/jruby#6607]
We have various places in our javadoc that are rejected by the strict javadoc generator in Java 11. We should be able to build javadoc on all released versions of Java, but at least on 11.

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.7:jar (attach-javadocs) on project jruby-core: MavenReportException: Error while creating archive:
[ERROR] Exit code: 1 - /Users/headius/projects/jruby-9.2/tmp/jruby-dist-build/core/src/main/java/org/jruby/RubyUnboundMethod.java:45: error: unknown tag: note
[ERROR]  * @note This was renamed from UnboundMethod.java
[ERROR]    ^
[ERROR] /Users/headius/projects/jruby-9.2/tmp/jruby-dist-build/core/src/main/java/org/jruby/embed/LocalVariableBehavior.java:34: error: malformed HTML
[ERROR]  * @author Yoko Harada <yokolet@gmail.com>
[ERROR]                        ^
[ERROR] /Users/headius/projects/jruby-9.2/tmp/jruby-dist-build/core/src/main/java/org/jruby/embed/LocalVariableBehavior.java:34: error: bad use of '>'
[ERROR]  * @author Yoko Harada <yokolet@gmail.com>
[ERROR]                                          ^
[ERROR] /Users/headius/projects/jruby-9.2/tmp/jruby-dist-build/core/src/main/java/org/jruby/embed/EvalFailedException.java:35: error: malformed HTML
[ERROR]  * @author Yoko Harada <yokolet@gmail.com>
[ERROR]                        ^
[ERROR] /Users/headius/projects/jruby-9.2/tmp/jruby-dist-build/core/src/main/java/org/jruby/embed/EvalFailedException.java:35: error: bad use of '>'
[ERROR]  * @author Yoko Harada <yokolet@gmail.com>
...
```

There are several pages of minor things to fix in the javadocs.

**Comment by headius** [comment]
I started cleaning this up and every set of fixes just opened up more things that were wrong. The javadoc linter penalizes you for partial documentation but ignores methods with no documentation at all. I will be looking into disabling the linter in the short term and doing a larger intentional javadoc update effort after 9.3 is released.

**Comment by headius** [comment]
@edipofederle I think I just abandoned the work when I discovered the JDK bug. We definitely want to revisit this now that the bug is fixed!

**Comment by headius** [comment]
This was mostly fixed by @rcrews in #7765, and I made a small additional fix in #7808. We do not yet fully modularize properly, but that will really have to wait until we can drop Java 8 due to the complexity of maintaining two completely different deployment structures.
